core/app/models/spree/store_credit.rb

```diff
@@ -288,4 +289,11 @@ def associate_credit_type
       self.credit_type = Spree::StoreCreditType.find_by_name(credit_type_name)
     end
   end
+
+  def apply_to_orders_in_confirm_state
```

Comment:
The logic only applies to orders in the confirm state because any order in a state prior to confirm will have store credit applied during [this transition](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order/checkout.rb#L84).